terraform/load_balancer.tf

```diff
@@ -0,0 +1,30 @@
+resource "digitalocean_loadbalancer" "lb" {
+  name                     = "lb-1"
+  region                   = "ams3"
+  enable_backend_keepalive = true
+  droplet_tag              = "web-lb"
+
+  forwarding_rule {
+    entry_port      = 80
+    entry_protocol  = "http"
+
+    target_port     = 80
+    target_protocol = "http"
+  }
+
+  sticky_sessions {
+    type               = "cookies"
+    cookie_name        = "DO-LB"
+    cookie_ttl_seconds = 300
+  }
```

Comment:
Is it possible to use the cookie that minitwit creates instead? Otherwise the user might log in on one instance and then 5 minutes later be assigned to the other server and no longer appear to be signed in.
Alternatively we could put cookies in the db which is maybe a more scale able solution?

Reply:
That is the exact scenario/problem that using a sticky session fixes though. It's how DigitalOcean says you need to do it: https://docs.digitalocean.com/products/networking/load-balancers/ Basically it always forwards the user to the same server once the DO-LB cookie is set. I don't think using the application cookie would work because the load balancer removes it before it reaches the web servers.

Reply:
The problem I see is that when the cookie for the session expires then we can't control if the user ends up in the correct server or a different one without the minitwit session.

Reply:
I'm not sure what you mean by that. The load balancer assigns a cookie to each session and that way knows which server to send it to. That's what we want, no? If we remove it, that's when all requests will be divided randomly among servers. Like I said, I don't think it's possible to use the minitwit cookie because it cannot go through the load balancer.

Reply:
Ah I see what you're saying now, the TTL is kinda low. Maybe we can increase it?

Reply:
I'm wondering if there can happen some weirdness if we recreate or add new servers. We can also just go with this setup and add a issue about moving minitwit cookies into the db.

Reply:
I don't think so, the load balancer is connected to its droplets through the tag, so if we add new servers they just need to be assigned that tag too, and it should work. But yeah like you say, maybe the cookie stuff would fit better in a new issue.